src/app/qgisapp.cpp

```diff
@@ -9730,7 +9731,8 @@ void QgisApp::duplicateLayers( const QList<QgsMapLayer *> &lyrList )
     // duplicate the layer style
     QString errMsg;
     QDomDocument style;
-    selectedLyr->exportNamedStyle( style, errMsg );
+    QgsReadWriteContext context;
+    selectedLyr->exportNamedStyle( style, errMsg, context );
```

Comment:
Why are these changes needed?

Reply:
exportNamedStyle was currently using a context created in QgsMapLayer. I think it should be created from the app to be able to report messages back.
Even if not used at the moment, this makes the code as the methods were initially thought.